pkg/wallet/wallet.go

```diff
@@ -182,9 +183,41 @@ func New(nickname string) (*Wallet, error) {
 
 	addr := blake3.Sum256(pubKey)
 
+	return CreateWalletFromKeys(nickname, privKey, pubKey, addr)
+}
+
+func Imported(nickname string, privateKey string) (*Wallet, error) {
+	privKeyBytes, _, err := base58.VersionedCheckDecode(privateKey[1:])
+	if err != nil {
+		return nil, fmt.Errorf("encoding private key B58: %w", err)
+	}
+
+	keypair := ed25519.NewKeyFromSeed(privKeyBytes)
+
+	pubKeyBytes := reflect.ValueOf(keypair.Public()).Bytes() // force conversion to byte array
+
+	addr := blake3.Sum256(pubKeyBytes)
+
+	return CreateWalletFromKeys(nickname, privKeyBytes, pubKeyBytes, addr)
+}
+
+func Delete(nickname string) (err error) {
+	err = os.Remove(GetWalletDirectory() + "wallet_" + nickname + ".json")
+	if err != nil {
+		return fmt.Errorf("deleting wallet 'wallet_%s.json': %w", nickname, err)
+	}
+
+	return nil
+}
+
+func AddressChecker(address string) bool {
+	return len(address) > MinAddressLength
+}
+
+func CreateWalletFromKeys(nickname string, privKeyBytes []byte, pubKeyBytes []byte, addr [32]byte) (*Wallet, error) {
 	var salt [16]byte
```

Comment:
the quick win is to initialize the salt array with random values
exmaple:
salt := [...]byte{0xb, 0xa, 0xb,0xe, 0xf}

Reply:
@gregLibert ?

Reply:
Why is there a need to initialize an Array?

If the array is not initialized at the time of declaration or any time after that then it will contain some random values in each memory position

Reply:
ok so fine !! i read somewhere is was initialized to zero by default.. but it seems its was wrong! sorry !! thats all for me ^^

Reply:
yes but what kind of random is this? It might be a bad pseudo random based on the current state of the memory. I think it is a concern if we need real randomness

Reply:
> yes but what kind of random is this? It might be a bad pseudo random based on the current state of the memory. I think it is a concern if we need real randomness

 I agree with that

Reply:
Maybe we should create an other issue for this topic ? Seems to be an important decision